src/object-store/src/layers/prometheus.rs

```diff
@@ -156,24 +156,27 @@ impl<A: Accessor> LayeredAccessor for PrometheusAccessor<A> {
         let timer = REQUESTS_DURATION_SECONDS
             .with_label_values(&[&self.scheme, Operation::Read.into_static()])
             .start_timer();
+        let read_res = self.inner.read(path, args).await;
+        timer.observe_duration();
```

Comment:
I count two parts duration here, one is creating the writer which is collected here, another is the writer future write part, which will be collected in the `PrometheusMetricWrapper::drop()`.

If `self.inner.read` return `Err` and I pass the timer inside `PrometheusMetricWrapper`, then no metric will be collected.

Reply:
> I count two parts duration here

It might make the duration metrics confusing as what we want to observe is the total cost to read or write an object.

> If self.inner.read return Err and I try passing the timer inside PrometheusMetricWrapper, then PrometheusMetricWrapper::new is not called and no metric will be collected. Is this right?

The timer itself is RAII so this doesn't matter.
https://docs.rs/prometheus/latest/prometheus/struct.HistogramTimer.html